h2o-core/src/main/java/water/Key.java

```diff
@@ -467,12 +472,13 @@ public String valueClassSimple() {
   }
 
   @Override public int hashCode() { return _hash; }
+
+  public boolean equals(Key k) {
+    return k != null && _hash == k._hash && Arrays.equals(k._kb,_kb);
+  }
+
   @Override public boolean equals( Object o ) {
-    if( this == o ) return true;
-    if( o == null ) return false;
-    Key k = (Key)o;
-    if( _hash != k._hash ) return false;
-    return Arrays.equals(k._kb,_kb);
+    return this == o || (o != null && (o instanceof Key) && equals((Key)o));
```

Comment:
Yes, of course, it's supposed to call the other equals. I just don't count on it; no big deal.
This method will return true if called like this:
    Object key3 = Key.make(s1);
    assertTrue(key1.equals(key3));